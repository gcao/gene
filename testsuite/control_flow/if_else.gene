#!/usr/bin/env gene run
# TEST: if/else expressions (requires 'else' keyword)

# Simple if with else keyword
(println "if true: " (if true "yes" else "no"))
(println "if false: " (if false "yes" else "no"))

# if without else - commented out as it causes segfault
# (println "if true no else: " (if true "only if true"))
# (println "if false no else: " (if false "should not print"))

# Nested if - commented out (causes infinite loop)
# (var y 10)
# (println "nested if: " 
#   (if (.< y 5)
#     "less than 5"
#   else
#     (if (.< y 15)
#       "between 5 and 15"
#     else
#       "greater than 15")))