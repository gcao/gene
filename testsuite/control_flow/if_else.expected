if true:  yes
if false:  no